apps/api/src/app/integrations/e2e/create-integration.e2e.ts

```diff
@@ -48,57 +49,120 @@ describe('Create Integration - /integration (POST)', function () {
-      providerId: 'sendgrid',
-      channel: 'email',
+      name: EmailProviderIdEnum.SendGrid,
+      providerId: EmailProviderIdEnum.SendGrid,
+      channel: ChannelTypeEnum.EMAIL,
       credentials: { apiKey: '123', secretKey: 'abc' },
       active: true,
       check: false,
     };
 
-    const secondInsertResponse = await insertIntegrationTwice(session, payload, false);
+    await insertIntegrationTwice(session, payload, false);
 
-    expect(secondInsertResponse.body.message).to.contain('Duplicate key');
+    const integrations = (await session.testAgent.get(`/v1/integrations`)).body.data;
+
+    const sendgridIntegrations = integrations.filter(
+      (integration) =>
+        integration.channel === payload.channel &&
+        integration._environmentId === session.environment._id &&
+        integration.providerId === EmailProviderIdEnum.SendGrid
+    );
+
+    expect(sendgridIntegrations.length).to.eql(2);
+
+    for (const integration of sendgridIntegrations) {
+      expect(integration.name).to.equal(payload.name);
+      expect(integration.identifier).to.exist;
+      expect(integration.providerId).to.equal(EmailProviderIdEnum.SendGrid);
+      expect(integration.channel).to.equal(ChannelTypeEnum.EMAIL);
+      expect(integration.credentials.apiKey).to.equal(payload.credentials.apiKey);
+      expect(integration.credentials.secretKey).to.equal(payload.credentials.secretKey);
+      expect(integration.active).to.equal(payload.active);
+    }
   });
 
-  it('should fail due missing param', async function () {
+  it('should not allow to create integration with same identifier', async function () {
     const payload = {
-      providerId: 'sendgrid',
-      channel: 'email',
-      active: true,
+      providerId: EmailProviderIdEnum.SendGrid,
+      channel: ChannelTypeEnum.EMAIL,
+      identifier: 'identifier',
+      active: false,
       check: false,
     };
+    await integrationRepository.create({
+      name: 'Test',
+      identifier: payload.identifier,
+      providerId: EmailProviderIdEnum.SendGrid,
+      channel: ChannelTypeEnum.EMAIL,
+      active: false,
+      _organizationId: session.organization._id,
+      _environmentId: session.environment._id,
+    });
 
-    const body = (await session.testAgent.post('/v1/integrations').send(payload)).body.data as IntegrationEntity;
+    const { body } = await session.testAgent.post('/v1/integrations').send(payload);
 
-    expect(body).to.equal(undefined);
+    expect(body.statusCode).to.equal(400);
+    expect(body.message).to.equal('Integration with identifier already exists');
   });
 
-  it('should deactivated old providers', async function () {
+  it('should not allow to activate the integration without the credentials', async function () {
     const payload = {
-      providerId: 'mailgun',
-      channel: 'email',
-      credentials: { apiKey: '123', secretKey: 'abc' },
+      providerId: EmailProviderIdEnum.SendGrid,
+      channel: ChannelTypeEnum.EMAIL,
```

Comment:
Here we have a problem, at least of coordination. We are stating credentials are optional in the creation of the request but if active is true then they are mandatory.
As to make the validation very complicated and to show it automatically with Swagger even more, I think we need to clarify this in the description of the endpoint and the property validation so it is documented in the API.